core/src/ast/data.rs

```diff
@@ -130,10 +130,10 @@ impl<V: FromVariant, F: FromField> Data<V, F> {
                     }
                 }
 
-                if !errors.is_empty() {
-                    Err(Error::multiple(errors))
-                } else {
+                if errors.is_empty() {
```

Comment:
The use of `!` here is deliberate to indicate that the error case is an "early" return - there are lots of cases of this that clippy didn't catch because there isn't an `else` block right after. I'm not sure it's worth making these ones different to satisfy clippy.

This might be a reason to revisit #115

Reply:
Your repo, your call! I'm happy to oblige, my idea was simply to implement all clippy lints